RevenueCatUI/Templates/V2/Components/Packages/PurchaseButton/PurchaseButtonComponentView.swift

```diff
@@ -83,6 +76,82 @@ struct PurchaseButtonComponentView: View {
         }
     }
 
+    private func purchase() async throws {
+        guard let action = self.viewModel.action else {
+            try await self.purchaseInApp()
+            return
+        }
+
+        switch action {
+        case .inAppCheckout:
+            try await self.purchaseInApp()
+        case .webCheckout:
+            try await self.purchaseWeb()
+        case .webProductSelection:
+            try await self.purchaseWebProductSelection()
+        }
+    }
+
+    private func purchaseInApp() async throws {
+        self.logIfInPreview(package: self.packageContext.package)
+
+        guard !self.purchaseHandler.actionInProgress else { return }
+
+        // WIP: Need to log warning if currently subscribed
+        guard let selectedPackage = self.packageContext.package
+                //                    , selectedPackage.currentlySubscribed
+        else {
+            Logger.warning(Strings.product_already_subscribed)
+            return
+        }
+
+        _ = try await self.purchaseHandler.purchase(package: selectedPackage)
+    }
+
+    private func purchaseWeb() async throws {
+        self.logIfInPreview(package: self.packageContext.package)
+
+        guard let selectedPackage = self.packageContext.package else {
+            Logger.warning(Strings.product_already_subscribed)
+            return
+        }
+
+        guard let webCheckoutUrl = self.packageContext.package?.webCheckoutUrl else {
+            Logger.warning(Strings.no_web_checkout_url_found)
```

Comment:
more like `Logger.error`, right? At this point we're in `purchaseWeb` but we're not able to initiate a web purchase, something went wrong